services/blockchain-indexer/app.js

```diff
@@ -49,33 +53,66 @@ const app = Microservice({
 	dependencies: [
 		'connector',
 	],
-});
+};
 
-setAppContext(app);
+const app = Microservice(defaultBrokerConfig);
 
-// Add routes, events & jobs
-if (config.operations.isIndexingModeEnabled) {
-	app.addMethods(path.join(__dirname, 'methods', 'indexer'));
-	app.addEvents(path.join(__dirname, 'events'));
-}
+// Start a temporary broker to query for SDK module names
+// To be used for dynamically registering the available module specific endpoints
+const tempApp = Microservice({
+	...defaultBrokerConfig,
+	name: 'temp_service_indexer',
+	events: {},
+});
+setAppContext(tempApp);
```

Comment:
Can we please refactor the code to execute this logic only when `config.operations.isDataRetrievalModeEnabled` is set.

Reply:
Applied. Thanks for the optimization 🙏